ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -176,7 +176,9 @@ public PlanNode getRight() {
   @Override
   public SchemaKStream<?> buildStream(final PlanBuildContext buildContext) {
 
-    ensureMatchingPartitionCounts(buildContext.getServiceContext().getTopicClient());
+    if (!joinKey.isForeignKey()) {
```

Comment:
Can we add a QTT for this?

Btw: do we already have a QTT for different key formats that we should also support for FK-joins.

Reply:
See PR description -- we already have a QTT for this but we can't enable it until the QTT fixes are in: https://github.com/confluentinc/ksql/blob/master/ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json#L659